### ISSUE TITLE
ENH: new orient setting for read_json to support common API format

### ISSUE THREAD

**Issue by ryancasburn-KAI** [pandas-dev/pandas#39913]
#### Is your feature request related to a problem?

I see many APIs return results in the form:

```json
{
"results": [
    {"a" :1,
     "b":2
    },
    {
     "a" :3,
     "b":4
    }
  ]
}
```
This format isn't directly supported by pandas. The data is in the "records" orient, but there is an extra layer. Currently to load this file I am using the requests module to load from https, then using the json module to strip out the outer layer, then feeding this data to pd.read_json as text. This feels like overkill, since pandas can read from https, but for this format (which is common for APIs) I need multiple other packages and lines of code.

This will change a three import, multi-line issue into a single import, single line solution.

#### Describe the solution you'd like

While I initially describe this as a new orient, I don't think that is the best way to implement this. I believe the `read_json` function should have a new parameter (such as "strip_layer") which will be the value of that outer layer. In the example above that would be "results". I make this suggestion as what is inside the outer layer could be several different orients, so we need to leave that as a possibility. This is something that happens first, then the data is processed.

#### API breaking implications

Need to consider what this means for chunking.

#### Additional context

My current code:

```python
import pandas as pd
import requests
import json

data = requests.get(url).json()
data = data["results"]
data = json.dumps(data)
data = pd.read_json(data)
```

versus my desired code with this improvement:

```python
import pandas as pd

data = pd.read_json(url, strip_layer="results")
```

**Comment by attack68** [comment]
is this about just wrapping the json output into a new json dict that has the key `results`. For example you can orient json in different ways: `records` `table` etc. so would you have to 'resultify' each type or have I misunderstood?

**Comment by attack68** [comment]
you don't have to dump and re-read, what about:


```
data = requests.get(url).json()
data = pd.DataFrame(data["results"])
```

But I see your point..

**Comment by ryancasburn-KAI** [comment]
True, I wasn't the most efficient in my example. Still need another package either way.

**Comment by jbrockmendel** [comment]
Having to use `requests` doesn't seem like a huge burden to me
